src/app/app.module.spec.ts

```diff
@@ -1,10 +1,181 @@
         let result = HttpLoaderFactory(null);
         expect(result).toEqual(new TranslateHttpLoader(null, "assets/i18n/"));
     });
+
+    it("should make a call to style the status bar light, prepare permissions, settings, translations, hockeyappintegration and firstpage", (done) => {
+        spyOn(comp, "ngOnInit").and.callThrough();
+        spyOn(statusBar, "styleLightContent");
+        spyOn(comp, "preparePermissions");
+        spyOn(comp, "prepareSettings");
+        spyOn(comp, "prepareHockeyAppIntegration");
+        spyOn(comp, "prepareFirstPage");
+
+        comp.ngOnInit().then(data => {
+            expect(statusBar.styleLightContent).toHaveBeenCalled();
+            expect(comp.preparePermissions).toHaveBeenCalled();
+            expect(comp.prepareSettings).toHaveBeenCalled();
+            expect(comp.prepareHockeyAppIntegration).toHaveBeenCalled();
+            expect(comp.prepareFirstPage).toHaveBeenCalled();
+            done();
+        });
+    });
+
+    it("should prepare the permissions for android", () => {
+        spyOn(platform, "is").and.callFake((arg) => {
+            if (arg === "android") return true; else return false;
+        });
+        spyOn(androidPermissions, "requestPermissions");
+
+        comp.preparePermissions();
+
+        expect(androidPermissions.requestPermissions).toHaveBeenCalledWith([androidPermissions.PERMISSION.READ_EXTERNAL_STORAGE, androidPermissions.PERMISSION.WRITE_EXTERNAL_STORAGE]);
+    });
+
+    it("should not request android permissions", () => {
+        spyOn(platform, "is").and.returnValue(false);
+        spyOn(androidPermissions, "requestPermissions");
+
+        comp.preparePermissions();
+
+        expect(androidPermissions.requestPermissions).not.toHaveBeenCalled();
+    });
+
+    it("should set the proper default language and theme", (done) => {
+        spyOn(settingsService, "getActiveTheme").and.callThrough();
+        spyOn(settingsService, "getLanguageSettings").and.callThrough();
+        spyOn(settingsService, "setActiveTheme");
+        spyOn(translate, "setDefaultLang");
+        spyOn(translate, "use");
+
+        comp.prepareSettings().then(data => {
+            expect(settingsService.getActiveTheme).toHaveBeenCalled();
+            expect(translate.setDefaultLang).toHaveBeenCalledWith("en");
+            expect(translate.use).toHaveBeenCalledWith("en");
+            expect(settingsService.setActiveTheme).toHaveBeenCalledWith("light-theme");
+            done();
+        });
+    });
+
+    it("should call hockey start with the right arguments and return a resolved promise", (done) => {
+        spyOn(hockeyApp, "start").and.returnValue(Promise.resolve());
+
+        comp.prepareHockeyAppIntegration().then(data => {
+            expect(hockeyApp.start).toHaveBeenCalledWith(jasmine.any(String), jasmine.any(String), true, true);
+            done();
+        });
+    });
+
+    it("should call hockey start and receive a rejected promise", (done) => {
```

Comment:
I am uncertain what exactly is being tested here. Since the above test case already asserts a promise is returned testing if a promise reject can be returned seems a bit redundant.

A better test would test if the promise reject is correctly handled by the calling function. However since we don't do this in our code even this test would not be needed.